src/graphql/node/control.rs

```diff
@@ -99,37 +99,60 @@ impl NodeControlMutation {
         }
 
         if let Some(ref target_agents) = apply_scope.agents {
-            let agent_manager = ctx.data::<BoxedAgentManager>()?;
-            if let Err(e) = notify_agents(
-                agent_manager,
-                &target_agents.updates,
-                &target_agents.disables,
-            )
-            .await
-            {
-                error!("Failed to notify agents for node {i} to be updated. However, the failure does not affect the node application operation.\nDetails:\n{e:?}",);
-            }
+            let hostname = {
+                let store = crate::graphql::get_store(ctx).await?;
+                let node_map = store.node_map();
+                let (node, _) = node_map.get_by_id(i)?.ok_or_else(|| {
+                    async_graphql::Error::new(format!("Node with ID {i} not found"))
+                })?;
+                node.profile.map(|p| p.hostname).unwrap_or_default()
+            };
 
-            info!(
-                "[{}] Node ID {i} - Node's agents are notified to be updated.\n{:?}",
-                chrono::Utc::now(),
-                node.agents.iter().filter_map(|agent| {
-                    if target_agents.updates.contains(&agent.key.as_str()) {
-                        Some(format!(
-                            "\nAgent key: {}, Config: {:?}, Draft: {:?}",
-                            agent.key, agent.config, agent.draft
-                        ))
-                    } else {
-                        None
-                    }
-                })
-            );
+            if hostname.is_empty() {
+                info!(
+                    "Node ID {i} - Node's agents are not notified because the hostname is empty."
```

Comment:
It seems to me that this results in no error being returned to the user, even with this message. Could you explain why?

Reply:
It is treated the same as we treat the failures of broadcast customer change and agent notification, which does not return error but only leave a log.

Reply:
Got it. I forgot that we decided to do it this way.